### PR TITLE
Added Windows os_version Virtual Table columns

### PR DESCRIPTION
**Pull request by PoppySeedPlehzr**

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit 343b1f19e2708d37a003ce26217cb900302ae87b (Job results: [603](https://jenkins.osquery.io/job/osqueryPullRequestAudit/603/)) failed the code audit.

**Comment by PoppySeedPlehzr** [comment]
Ah. It seems that indeed the `GetVersionEx` function was indeed deprecated in such a way that it doesn't work natively for Windows 8+.  I'm going to update this PR to make use of the WMI tables again to pull the correct Major, Minor, and Build versions, but it seems we might still be able to leverage the `GetVersionEx` function for a few other components.

**Comment by osqueryer** [comment]
:no_good: The commit 0fc64d2f49f8e9b80643c034844e9d2165a5cc50 (Job results: [605](https://jenkins.osquery.io/job/osqueryPullRequestAudit/605/)) failed the code audit.

**Comment by osqueryer** [comment]
:no_good: The commit caaf264cb869cab954b4befb8b7bb43970bfd83f (Job results: [606](https://jenkins.osquery.io/job/osqueryPullRequestAudit/606/)) failed the code audit.

File: osquery/tables/system/windows/os_version.cpp [hunk 1]
```diff
@@ -8,42 +8,171 @@
...
+
+#pragma warning(suppress : 4996)
+  GetVersionEx(reinterpret_cast<OSVERSIONINFO*>(&osvi));
+
+  r["major"] = INTEGER(osvi.dwMajorVersion);
+  r["minor"] = INTEGER(osvi.dwMinorVersion);
+  r["build"] = INTEGER(osvi.dwBuildNumber);
+  r["patch"] = INTEGER(osvi.wServicePackMajor);
+  r["version"] = std::to_string(osvi.dwMajorVersion) + "." +
+                 std::to_string(osvi.dwMinorVersion);
```
Comment (PoppySeedPlehzr) [inline osquery/tables/system/windows/os_version.cpp]: Yeah I believe so, adding that now.

File: osquery/tables/system/windows/os_version.cpp [hunk 2]
```diff
@@ -44,6 +157,23 @@ QueryData genOSVersion(QueryContext& context) {
   default:
     break;
   }
+
+  r["platform"] = "windows";
+  r["platform_like"] = "windows";
+  r["version"] = r["major"] + "." + r["minor"] + "." + r["build"];
+  if (version.size() >= 2) {
```
Comment (PoppySeedPlehzr) [inline osquery/tables/system/windows/os_version.cpp]: I think this check is really redundant. I only dropped it to be safe and ensure the call to `GetProductInfo` has values. From what I can tell, every version of windows will _always_ have a major, minor, and build number.

**Pull request merged**